protocols/relay/CHANGELOG.md

```diff
@@ -11,7 +11,10 @@
 
 - Update to `libp2p-swarm` `v0.42.0`.
 
+- Optimize listeners polling behaviour in the tcp transport, fixing [2781](https://github.com/libp2p/rust-libp2p/issues/2781). See [PR 3361].
```

Comment:
Why are we making a changelog entry for relay when we are changing code in TCP?

Reply:
Btw, I don't think this needs a changelog entry at all because it is an internal refactoring.

Reply:
Fixed. The PR templates explicitly asks to make a changelog entry. I am not making the rules :)

Reply:
Fair! Most PRs, especially from external contributors add functionality or fix bugs, both of which are user-facing. This one is the exception to the rule :)